test/test_pickle.py

```diff
@@ -59,3 +63,31 @@ def test_pickle_hmcecs():
     mcmc.run(random.PRNGKey(0))
     pickled_mcmc = pickle.loads(pickle.dumps(mcmc))
     test_util.check_close(mcmc.get_samples(), pickled_mcmc.get_samples())
+
+
+def poisson_regression(x, N):
+    rate = numpyro.sample("param", dist.Gamma(1.0, 1.0))
+    batch_size = len(x) if x is not None else None
+    with numpyro.plate("batch", N, batch_size):
+        numpyro.sample("x", dist.Poisson(rate), obs=x)
+
+
+@pytest.mark.parametrize("guide_class", [AutoDelta, AutoDiagonalNormal, AutoNormal])
+def test_pickle_autoguide(guide_class):
+    x = np.random.poisson(1.0, size=(100,))
+
+    guide = guide_class(poisson_regression)
+    optim = numpyro.optim.Adam(1e-2)
+    svi = SVI(poisson_regression, guide, optim, numpyro.infer.Trace_ELBO())
+    svi_result = svi.run(random.PRNGKey(1), 3, x, len(x))
+    pickled_guide = pickle.loads(pickle.dumps(guide))
+
+    predictive = Predictive(
+        poisson_regression,
+        guide=pickled_guide,
+        params=svi_result.params,
+        num_samples=1,
+        return_sites=["param", "x"],
+    )
+    samples = predictive(random.PRNGKey(1), None, 1)
+    assert set(samples.keys()) == {"param", "x"}
```

Comment:
Maybe better to assert that `pickled_guide` returns the same samples as `guide` instead of just verifying that all sites are present?